src/components/sbb-toast/sbb-toast.tsx

```diff
@@ -0,0 +1,260 @@
+
+  @Element() private _element!: HTMLElement;
+
+  private _handlerRepository = new HandlerRepository(
+    this._element,
+    languageChangeHandlerAspect((l) => (this._currentLanguage = l)),
+    namedSlotChangeHandlerAspect((m) => (this._namedSlots = m(this._namedSlots)))
+  );
+
+  private _closeTimeout: ReturnType<typeof setTimeout>;
+
+  /**
+   * Role of the live region. This is only for Firefox as there is a known issue where Firefox +
+   * JAWS does not read out aria-live message.
+   */
+  private get _role(): ToastAriaRole {
+    if (!isFirefox()) {
+      return;
+    }
+
+    if (this.politeness === 'polite') {
+      return 'status';
+    } else if (this.politeness === 'assertive') {
+      return 'alert';
+    }
+  }
+
+  /**
+   * Open the toast.
+   * If there are other opened toasts in the page, close them first.
+   */
+  @Method() public async open(): Promise<void> {
+    if (this._state !== 'closed') {
+      return;
+    }
+
+    this._closeOtherToasts();
+
+    this._state = 'opening';
+    this.willOpen.emit();
+  }
+
+  /**
+   * Close the toast.
+   */
+  @Method() public async close(): Promise<void> {
+    if (this._state !== 'opened') {
+      return;
+    }
+
+    clearTimeout(this._closeTimeout);
+
+    this._state = 'closing';
+    this.willClose.emit();
+  }
+
+  @Method() public async getState(): Promise<SbbOverlayState> {
+    return this._state;
+  }
+
+  public connectedCallback(): void {
+    this._handlerRepository.connect();
+  }
+
+  public disconnectedCallback(): void {
+    clearTimeout(this._closeTimeout);
+    this._handlerRepository.disconnect();
+  }
+
+  private _onActionSlotChange(event: Event): void {
+    const slotNodes = (event.target as HTMLSlotElement).assignedNodes();
+
+    // Force the visual state on slotted buttons
+    slotNodes
+      .filter((el) => el.nodeName === 'SBB-BUTTON')
+      .forEach((btn: HTMLSbbButtonElement) => {
+        btn.variant = 'transparent';
+        btn.negative = true;
+        btn.size = 'm';
+      });
+
+    // Force the visual state on slotted links
+    slotNodes
+      .filter((el) => el.nodeName === 'SBB-LINK')
+      .forEach((link: HTMLSbbLinkElement) => {
+        link.variant = 'inline';
+        link.negative = true;
+      });
+  }
+
+  private _onToastAnimationEnd(event: AnimationEvent): void {
+    // On toast opened
+    if (event.animationName === 'open') {
+      this._state = 'opened';
+      this.didOpen.emit();
+
+      // Start the countdown to close it
+      if (this.timeout) {
+        this._closeTimeout = setTimeout(() => this.close(), this.timeout);
+      }
+    }
+
+    // On toast closed
+    if (event.animationName === 'close') {
+      this._state = 'closed';
+      this.didClose.emit();
+    }
+  }
+
+  /**
+   * Since we do not stack toasts, we force the closing on other existing opened toasts
+   */
+  private async _closeOtherToasts(): Promise<void> {
+    document.querySelectorAll('sbb-toast').forEach(async (t) => {
```

Comment:
Initially, I thought if we should support toasts in shadow DOM but, in the end, I couldn't find a realistic case where this would be needed. But, since you pointed it out, it doesn't hurt to handle those cases too. I will apply your solution :)

Reply:
Hello @kyubisation, 
unfortunately, the "WeakRef" class has been added in ES2021.
Since we are targeting the ES2017, we need to add a reference to a part of the new lib.
I don't fully understand the consequences of this change, do you see a problem that I don't?

![image](https://github.com/lyne-design-system/lyne-components/assets/12052575/a1e2eee6-4d92-44d3-93c0-a6f5ad2d557c)